MaxText/max_utils.py

```diff
@@ -680,3 +669,44 @@ def summarize_pytree_data(params, name="Params", raw=False):
             f"\tTotal memory usage: {total_param_size:.3f} bytes \n"
             f"\tAvg size: {avg_param_size:.3f} bytes\n")
   return num_params, total_param_size, avg_param_size
+
+
+class Profiler:
+  """Activate/deactivate a profiler based on the 'profiler' config"""
+
+  def __init__(self, config, optional_postfix=""):
+    self.libcudart = None
+    self.mode = config.profiler
+    if self.mode == "" and config.enable_profiler:
+      self.mode = "xprof"
+    self.upload_all_profiler_results = config.upload_all_profiler_results
+    self.output_path = os.path.join(config.tensorboard_dir, optional_postfix)
+
+  def activate(self):
+    """Start the profiler.
+    nsys profiler becomes no-op when libcudart.so is not available on the system"""
+    if self.mode == "nsys" and (self.upload_all_profiler_results or jax.process_index() == 0):
```

Comment:
nit: Shall we add a check at the beginning? 
`if not (self.upload_all_profiler_results or jax.process_index() == 0) return` 

Thus, we don't need to add this check for each mode (especially when we introduce other types of profiles in the future).

Reply:
Agreed this check is probably universal for any profiler